zingolib/src/wallet/keys/unified.rs

```diff
@@ -5,33 +5,65 @@ use std::{
 
 use bip0039::Mnemonic;
 use byteorder::{ReadBytesExt, WriteBytesExt};
-use orchard::keys::Scope;
+use orchard::keys::{FullViewingKey as OrchardFvk, Scope};
 
+use zcash_address::unified::{Container, Encoding, Fvk, Ufvk};
 use zcash_client_backend::address::UnifiedAddress;
 use zcash_encoding::Vector;
-use zcash_primitives::{legacy::TransparentAddress, zip32::DiversifierIndex};
+use zcash_primitives::{
+    legacy::TransparentAddress, zip32::DiversifierIndex,
+    zip32::ExtendedFullViewingKey as SaplingExtFvk,
+};
 use zingoconfig::ZingoConfig;
 
 use crate::wallet::traits::ReadableWriteable;
 
-use super::{extended_transparent::KeyIndex, get_zaddr_from_bip39seed, ToBase58Check};
+use super::{
+    extended_transparent::{ExtendedPubKey, KeyIndex},
+    get_zaddr_from_bip39seed, ToBase58Check,
+};
 
 #[derive(Clone, Debug)]
-pub struct UnifiedSpendCapability {
-    orchard_key: orchard::keys::SpendingKey,
-    sapling_key: zcash_primitives::zip32::ExtendedSpendingKey,
-    transparent_parent_key: super::extended_transparent::ExtendedPrivKey,
-    transparent_child_keys: Vec<(usize, secp256k1::SecretKey)>,
+pub enum Capability<ViewingKeyType, SpendKeyType> {
```

Comment:
@AloeareV proposes making this `#[non_exhaustive']` both the future proof against `SpendViaHWW(ViewingKeyType)`, and to future proof against allowing different types of viewing keys `Outgoing` `Incoming`.

Reply:
3a73b7f11c9338ac302895471672502f197eca96